trace/trace.go

```diff
@@ -157,11 +171,149 @@ func decodeHex(h string, b []byte) error {
 	return nil
 }
 
+// TraceState provides additional vendor-specific trace identification information
+// across different distributed tracing systems. It represents an immutable list consisting
+// of key/value pairs. There can be a maximum of 32 entries in the list.
+//
+// Key and value of each list member must be valid according to the W3C Trace Context specification
+// (see https://www.w3.org/TR/trace-context-1/#key and https://www.w3.org/TR/trace-context-1/#value
+// respectively).
+//
+// Trace state must be valid according to the W3C Trace Context specification at all times. All
+// mutating operations validate their input and, in case of valid parameters, return a new TraceState.
+type TraceState struct { //nolint:golint
```

Comment:
this is a nitpick, but this addition do this file might be easier to read in its own tracestate-specific file.

Reply:
Do you mean by that to put `TraceState` into a separate file?

Reply:
Yeah, `TraceState` and related things. But, we should probably do this in a separate PR as there are many good splits that can happen.